place_tracker/lib/place_details.dart

```diff
@@ -0,0 +1,215 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+
+import 'place.dart';
+
+class PlaceDetails extends StatefulWidget {
+  PlaceDetails(this._place);
+
+  final Place _place;
+
+  @override
+  PlaceDetailsState createState() => PlaceDetailsState(_place);
+}
+
+class PlaceDetailsState extends State<PlaceDetails> {
+  PlaceDetailsState(Place place) {
+    this._place = place.copy();
+  }
+
+  Place _place;
+  GoogleMapController mapController;
+  final TextEditingController _nameController = TextEditingController();
+  final TextEditingController _descriptionController = TextEditingController();
+
+  void onMapCreated(GoogleMapController controller) async {
+    mapController = controller;
+    mapController.addMarker(MarkerOptions(position: _place.latLng));
+  }
+
+  Widget _starBar() {
```

Comment:
This function, `_nameTextField`, `_descriptionTextField`, and `_map`, can be stateless widgets. For example:
```
class _StarBar extends StatelessWidget {
  const _StarBar({
    Key key,
    this.maxStars = 5;
    @required this.rating,
    @required this.onChanged;
  }) : /* TBD asserts */ super(key: key);

  final int maxStars;
  final int rating;
  final ValueChanged<int> onChanged;

  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.center,
      children: List.generate(maxStars, (int index) {
        IconButton(
          icon: Icon(Icons.star),
          iconSize: 40.0,
          color: rating >= index ? Colors.amber : Colors.grey[400],
          onPressed: () {
            onChanged(rating);
          },
        ),
      }).toList(),
    );
  }
}
```

Note also: it's best to eliminate centering etc wrapping around a leaf widget like this. It's easier to understand things if the parent adds that stuff. I've left out the `Center` in this case; maybe you want to create the Row with `mainAxisSize: MainAxisSize.min` here.

Reply:
In this example, what would the provided onChanged callback be?

_StarBar(rating: 5, onChanged: ???)

Reply:
Done.